### PR TITLE
fix(zod): allow custom markDefs to passthrough zod

### PR DESCRIPTION
**Pull request by mckelveygreg**
I just started using @sanity-typed/zod to parse responses, but I noticed it was stripping out custom markDefs from my `block` content. 

Adding `passthrough` keeps those fields from getting removed, and I'm not sure how else you would want to integrate this, as the block types seem to get very complicated!

I added some tests (though I had a hard time running them very well locally...)

### CODE REVIEW

**Comment by saiichihashimoto** [comment]
I haven't taken a look at this yet, will be getting to this soon!

File: packages/zod/src/block.test.ts [hunk 1]
```diff
@@ -188,6 +188,71 @@ describe("block", () => {
         InferSchemaValues<typeof config>["foo"][number]["children"]
       >();
     });
+    it("builds parser for custom markDefs", () => {
```
Comment (saiichihashimoto) [inline packages/zod/src/block.test.ts:191]: I'd change this to "allows custom markDefs". It's definitely not parsing them.

File: packages/zod/src/block.test.ts [hunk 2]
```diff
@@ -330,6 +395,64 @@ describe("block", () => {
 
       const parsed = zods.foo.parse(unparsed);
 
+      expect(parsed).toStrictEqual(unparsed);
+      expectType<(typeof parsed)["children"]>().toStrictEqual<
+        InferSchemaValues<typeof config>["foo"]["children"]
+      >();
+    });
+    it("builds parser for custom markDefs", () => {
```
Comment (saiichihashimoto) [inline packages/zod/src/block.test.ts:403]: Same.

**Review by saiichihashimoto — CHANGES_REQUESTED** [review]
Maybe `git commit --amend` to rerun linting, or just `npm run lint` and make another commit. I've added an eslint plugin that addresses some of the spacing comments I was going to make.

Also, looks like some of the existing tests might be failing.

**Comment by mckelveygreg** [comment]
Linting updated.

Yeah, making that a `.passthough()` seems to break the `InferSchemaValues`

**Comment by saiichihashimoto** [comment]
Yeah it looks like you'll need to see what `passthrough()` actually does to the types and add them to the corresponding `@sanity-typed/types` type for it to make sense.

**Comment by saiichihashimoto** [comment]
Once you're able to figure out what's the fix for `types` or for the zod parser to have them match, we should be good to go.

**Comment by mckelveygreg** [comment]
I haven't had a chance to troubleshoot. I have the dist patched so it will work in our project, but will have to come back to the types so we can have a proper fix 😞

**Comment by saiichihashimoto** [comment]
@mckelveygreg if I can fix the bugs in #535, we might actually have a more strongly typed solution than `passthrough()`, which actually enforces the types match what are in `marks.annotations`.

**Comment by saiichihashimoto** [comment]
@mckelveygreg Can you try `@sanity-typed/types@6.0.0` and `@sanity-typed/zod@2.0.0`? `markDefs` are now checked against the `marks.annotations` provided to the `block` definition, so the zod parser and types should correlate to your config instead of using `passthrough()`.

**Comment by mckelveygreg** [comment]
Hmm, i spent some time trying to get things updated, but I was getting infinite instantiation errors if I had anything in the `annotations` field (even with `definedArrayMember` and `defineField` just like in your new docs). 
I'll have to comeback later for updates 🤷 

Thanks for keeping me in the loop!
But it looks like this PR can probably close because you have taken a different direction?

**Comment by saiichihashimoto** [comment]
Would you be able to create an issue with your config that is creating the infinite instantiation issue so I can debug it?

**Comment by saiichihashimoto** [comment]
But yes, I'll close this!

**Pull request closed without merging**